oracle-linux-image-tools/bin/common.sh

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+
+# shellcheck disable=SC1090
+#
+# Create minimal Oracle Linux images
+#
+# Copyright (c) 2022 Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at
+# https://oss.oracle.com/licenses/upl.
+#
+# Description: creates minimal Oracle Linux images which can be used in cloud
+# environment.
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+#
+
+#######################################
+# Return 1 (true) if packaged using VirtualBox
+# Globals:
+#   PACKER_BUILDER
+# Arguments:
+#   None
+# Returns:
+#   1 -- packaged via VirtualBox
+#   0 -- packaged via QEMU
+#######################################
+common::is_vbox() {
+  test "${PACKER_BUILDER}" = "virtualbox-iso.x86-64"
+  return
+}
+
+#######################################
+# Generate manifest
+# Globals:
+#   None
+# Arguments:
+#   - files to include in the manifest
+# Returns:
+#   - writes manifest to stdout
+#######################################
+common::make_manifest() {
+  sha1sum "$@" | sed --regexp-extended 's/(.*) +(.*)/SHA1 (\2) = \1/g'
```

Comment:
From the `Open Virtualization Format Specification` document, there are no spaces around the file name:

> file_digest   = algorithm "(" file_name ")" "=" sp digest nl 

e.g.

```
SHA1(template.ovf)= bacdb0fa1c7e35f933c423066139d581caf1168d 
```

instead of

```
SHA1 (template.ovf) = bacdb0fa1c7e35f933c423066139d581caf1168d 
```

```suggestion
  sha1sum "$@" | sed --regexp-extended 's/(.*) +(.*)/SHA1(\2)= \1/g'
```

Reply:
As a small discourse, the standard creates syntactical ambiguity. If the filename starts / ends with a space, there is no reliable way to determine it: a space between parenthesis and filename is allowed too. Oh, well.

Reply:
Another interpretation point: "whitespace is allowed" -- it is not plural. so, `sha1 (foo) = 123` is legal. So is `sha1 (foo) =   123` (three spaces, one around `sp` in ABNF). But not `sha1   (foo)= 123`. :-)